cdc/sink/dispatcher/topic/dispatcher_test.go

```diff
@@ -0,0 +1,217 @@
+)
+
+func TestStaticTopicDispatcher(t *testing.T) {
+	row := &model.RowChangedEvent{
+		Table: &model.TableName{
+			Schema: "db1",
+			Table:  "tbl1",
+		},
+	}
+
+	ddl := &model.DDLEvent{
+		TableInfo: &model.SimpleTableInfo{
+			Schema: "db1",
+			Table:  "tbl1",
+		},
+	}
+
+	p := NewStaticTopicDispatcher("cdctest")
+	require.Equal(t, p.DispatchRowChangedEvent(row), "cdctest")
+	require.Equal(t, p.DispatchDDLEvent(ddl), "cdctest")
+}
+
+func TestDynamicTopicDispatcherForSchema(t *testing.T) {
+	t.Parallel()
+
+	topicExpr := Expression("hello_{schema}_world")
+	err := topicExpr.Validate()
+	require.Nil(t, err)
+	testCase := []struct {
+		row         *model.RowChangedEvent
+		expectTopic string
+	}{
+		{
+			row: &model.RowChangedEvent{
+				Table: &model.TableName{
+					Schema: "test1",
+					Table:  "tb1",
+				},
+			},
+			expectTopic: "hello_test1_world",
+		},
+		{
+			row: &model.RowChangedEvent{
+				Table: &model.TableName{
+					Schema: "test1",
+					Table:  "tb2",
+				},
+			},
+			expectTopic: "hello_test1_world",
+		},
+		{
+			row: &model.RowChangedEvent{
+				Table: &model.TableName{
+					Schema: "test2",
+					Table:  "tb1",
+				},
+			},
+			expectTopic: "hello_test2_world",
+		},
+		{
+			row: &model.RowChangedEvent{
+				Table: &model.TableName{
+					Schema: "test2",
+					Table:  "tb2",
+				},
+			},
+			expectTopic: "hello_test2_world",
+		},
+	}
+
+	p := NewDynamicTopicDispatcher("cdctest", topicExpr)
+	for _, tc := range testCase {
+		require.Equal(t, tc.expectTopic, p.DispatchRowChangedEvent(tc.row))
+	}
+}
+
+func TestDynamicTopicDispatcherForTable(t *testing.T) {
+	t.Parallel()
+
+	topicExpr := Expression("{schema}_{table}")
+	err := topicExpr.Validate()
+	require.Nil(t, err)
+	testCases := []struct {
+		row           *model.RowChangedEvent
+		expectedTopic string
+	}{
+		{
+			row: &model.RowChangedEvent{
+				Table: &model.TableName{
+					Schema: "test1",
+					Table:  "tb1",
+				},
+			},
+			expectedTopic: "test1_tb1",
+		},
+		{
+			row: &model.RowChangedEvent{
+				Table: &model.TableName{
+					Schema: "test1",
+					Table:  "tb2",
+				},
+			},
+			expectedTopic: "test1_tb2",
+		},
+		{
+			row: &model.RowChangedEvent{
+				Table: &model.TableName{
+					Schema: "test2",
+					Table:  "tb1",
+				},
+			},
+			expectedTopic: "test2_tb1",
+		},
+		{
+			row: &model.RowChangedEvent{
+				Table: &model.TableName{
+					Schema: "test2",
+					Table:  "tb2",
+				},
+			},
+			expectedTopic: "test2_tb2",
+		},
+	}
+	p := NewDynamicTopicDispatcher("cdctest", topicExpr)
+	for _, tc := range testCases {
+		require.Equal(t, tc.expectedTopic, p.DispatchRowChangedEvent(tc.row))
+	}
+}
+
+func TestDynamicTopicDispatcherForDDL(t *testing.T) {
+	t.Parallel()
+
+	defaultTopic := "cdctest"
+	topicExpr := Expression("{schema}_{table}")
+	testCases := []struct {
+		ddl           *model.DDLEvent
+		expectedTopic string
+	}{
+		{
+			ddl: &model.DDLEvent{
+				TableInfo: &model.SimpleTableInfo{
```

Comment:
What about `{Schema: "", Table: ""}`?

Reply:
>  What about `{Schema: "", Table: ""}`?

For what kind of DDL would the Schema and Table be empty?

Reply:
AFAIK, no such DDL, just in case 😀.